src/mame/audio/mw8080bw.cpp

```diff
@@ -727,46 +736,52 @@ void gunfight_audio_device::write(u8 data)
 	// the 74175 latches and inverts the top 4 bits
 	switch ((~data >> 4) & 0x0f)
 	{
-	case 0x00:
-		break;
 	case 0x01: // LEFT SHOOT sound (left speaker)
-		m_samples[0]->start(0, 0);
+		this->left_shot_w(1);
 		break;
 	case 0x02: // RIGHT SHOOT sound (right speaker)
-		m_samples[1]->start(0, 0);
+		this->right_shot_w(1);
 		break;
 	case 0x03: // LEFT HIT sound (left speaker)
-		m_samples[0]->start(0, 1);
+		this->left_hit_w(1);
 		break;
-	case 0x04: // enable RIGHT HIT sound (right speaker)
-		m_samples[1]->start(0, 1);
+	case 0x04: // RIGHT HIT sound (right speaker)
+		this->right_hit_w(1);
 		break;
-	default:
-		logerror("%s: Unknown sh port write %02x\n", machine().describe_context(), data);
+	default:   // any other value will turn off the sound switches
+		this->left_shot_w(0);
+		this->right_shot_w(0);
+		this->left_hit_w(0);
+		this->right_hit_w(0);
```

Comment:
What's the point of all the `this->` here?  What’s the point of having the trampolines at all?  Why not just set the lines directly?

Reply:
If that's a better way to do it, then that's fine. I'm just trying to use other drivers as a guideline for how to hook this up, and my knowledge of MAME internals is still far too limited to see that I wasn't following the best path here.